shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-env/src/test/java/org/apache/shardingsphere/test/integration/env/container/atomic/storage/config/impl/mysql/MySQLContainerConfigurationFactory.java

```diff
@@ -35,25 +36,53 @@ public final class MySQLContainerConfigurationFactory {
     
     /**
      * Create new instance of MySQL container configuration.
-     * 
+     *
      * @return created instance
      */
     public static StorageContainerConfiguration newInstance() {
-        return new StorageContainerConfiguration(getCommand(), getContainerEnvironments(), getMountedResources());
+        return new StorageContainerConfiguration(getDefaultCommand(), getDefaultContainerEnvironments(), getDefaultMountedResources());
     }
     
-    private static String getCommand() {
+    /**
+     * Create new instance of MySQL container configuration with parameter.
+     *
+     * @param command command
+     * @param containerEnvironments container environments
+     * @param mountedResources mounted resources
+     * @return created instance
+     */
+    public static StorageContainerConfiguration newInstance(final String command, final Map<String, String> containerEnvironments, final Map<String, String> mountedResources) {
+        return new StorageContainerConfiguration(ObjectUtils.defaultIfNull(command, getDefaultCommand()), ObjectUtils.defaultIfNull(containerEnvironments, getDefaultContainerEnvironments()),
+                ObjectUtils.defaultIfNull(mountedResources, getDefaultMountedResources()));
+    }
+    
+    /**
+     * Get default command.
+     *
+     * @return command
+     */
+    public static String getDefaultCommand() {
         return "--server-id=" + ContainerUtil.generateMySQLServerId();
     }
     
-    private static Map<String, String> getContainerEnvironments() {
+    /**
+     * Get default container environments.
+     *
+     * @return container environments
+     */
+    public static Map<String, String> getDefaultContainerEnvironments() {
         Map<String, String> result = new HashMap<>(2, 1);
         result.put("LANG", "C.UTF-8");
         result.put("MYSQL_RANDOM_ROOT_PASSWORD", "yes");
         return result;
     }
     
-    private static Map<String, String> getMountedResources() {
+    /**
+     * Get default mounted resources.
+     *
+     * @return container environments
+     */
+    public static Map<String, String> getDefaultMountedResources() {
```

Comment:
If these methods are just used in class itself, then it's not necessary to set as `public`

Reply:
Looks it might be better to refactor MySQLContainerConfigurationFactory to MySQLContainerConfiguration, it could be easier to extend it. And also other databases impl.
But it need too much changes, so just keep it for now.